crates/uv/src/lib.rs

```diff
@@ -1805,6 +1807,19 @@ async fn run_project(
             let args = settings::InitSettings::resolve(args, filesystem, environment);
             show_settings!(args);
 
+            // The `--project` arg is being deprecated for `init` with a warning now and an error in preview
+            if explicit_project {
+                if globals.preview.is_enabled(PreviewFeatures::DEPRECATE_PROJECT_FOR_INIT) {
+                    bail!(
+                        "The argument '--project' cannot be used for 'init'. Consider using `--directory` instead."
+                    );
+                } else {
+                    warn_user!(
+                        "The `--project` flag is deprecated for `uv init` and ignored in some cases. Consider using `--directory` to change the working directory instead."
+                    );
+                }
```

Comment:
The difference in language between these is confusing, e.g., we're switching between "argument" and "flag" here. I'd say

> The `--project` option cannot be used in `uv init`. Use `--directory` instead.

and

> Use of the `--project` option in `uv init` is deprecated and support will be removed in a future release . Consider using `--directory` instead.

Reply:
It is nice to have something about it being ignored, but "some cases" seems ambiguous. Can't we tell here if it will be ignored? Then we could say

> The `--project` option has no affect in `uv init` and will be removed in a future release. Consider...

Reply:
Can't we detect that case here though and only say it has no effect if there's also a positional path?

Reply:
That also suggests a different hint for that case, e.g., when they do `uv init --project foo` we should suggest `uv init foo` not `uv init --directory foo`, I think.

Reply:
Oh, yeah, I can definitely detect that case. Interesting about simplifying to `uv init foo`. I think that makes sense.